jscomp/syntax/src/res_core.ml

```diff
@@ -3923,6 +3941,36 @@ and parseListExpr ~startPos p =
             loc))
       [(Asttypes.Nolabel, Ast_helper.Exp.array ~loc listExprs)]
 
+and parseDictExpr ~startPos p =
+  let rows =
+    parseCommaDelimitedRegion ~grammar:Grammar.DictRows ~closing:Rbrace
+      ~f:parseDictExprRow p
+  in
+  let loc = mkLoc startPos p.endPos in
+  let toKeyValuePair
+      (recordItem : Longident.t Location.loc * Parsetree.expression) =
+    match recordItem with
+    | ( {Location.txt = Longident.Lident key; loc = keyLoc},
+        ({pexp_loc = valueLoc} as valueExpr) ) ->
+      Some
+        (Ast_helper.Exp.tuple
+           ~loc:(mkLoc keyLoc.loc_start valueLoc.loc_end)
+           [
+             Ast_helper.Exp.constant ~loc:keyLoc (Pconst_string (key, None));
+             valueExpr;
+           ])
+    | _ -> None
+  in
+  let keyValuePairs = List.filter_map toKeyValuePair rows in
+  Parser.expect Rbrace p;
+  Ast_helper.Exp.apply ~loc
+    (Ast_helper.Exp.ident ~loc
+       (Location.mkloc
+          (Longident.Ldot
+             (Longident.Ldot (Longident.Lident "Js", "Dict"), "fromArray"))
```

Comment:
Related https://github.com/rescript-lang/rescript-compiler/issues/6836

@cknitt I think this needs to be adapted, right?

Reply:
Right, but this can be done independently later.